src/core/socialprovider.js

```diff
@@ -302,6 +303,12 @@ XMPPSocialProvider.prototype.sendMessage = function(to, msg, continuation) {
             body += this.messages[to][i].message;
           }
           message.t(body);
+
```

Comment:
This block is only called if status !=== "ONLINE". Why are we sending messages to clients that are either OFFLINE or ONLINE_WITH_OTHER_APP?